health_rec/api/routes.py

```diff
@@ -151,3 +154,38 @@ async def get_services_count(
         The number of services.
     """
     return await chroma_service.get_services_count()
+
+
+@router.get("/rerank", response_model=List[Service])
+async def rerank_recommendations(
+    query: str, retrieval_k: int = 20, output_k: int = 5
+) -> Union[ServiceDocument | List[ServiceDocument]]:
+    """
+    Generate re-ranked list of services based on the input query.
+
+    Parameters
+    ----------
+    query : str
+        The user's input query.
+    retrieval_k : Optional[int]
+        Number of services to retrieve initially. Default is 10.
+    output_k : Optional[int]
+        Number of services to return after re-ranking. Default is 5.
+
+    Returns
+    -------
+    services: List[ServiceDocument]
+        A list of services ordered by relevance to the query.
+    """
+    try:
+        config = RerankingConfig(
+            retrieval_k=min(max(1, retrieval_k), 10),  # Limit between 1 and 20
+            output_k=min(
```

Comment:
Same here, checking if the output_k to be less than or equal to retrieval_k if not raise value error rather than hiding that from the user.